proxy/hdrs/HdrHeap.h

```diff
@@ -235,6 +235,23 @@ class HdrHeap
     }
   }
 
+  // Working function to copy strings into a new heap
+  // Unlike the HDR_MOVE_STR macro, this function will call
+  // allocate_str which will update the new_heap to create more space
+  // if there is not originally sufficient space
+  inline std::string_view
+  localize(const std::string_view &string)
+  {
+    if (string.length() > 0) {
+      char *new_str = this->allocate_str(string.length());
+      if (new_str) {
+        memcpy(new_str, string.data(), string.length());
+      }
+      return {new_str, string.length()};
```

Comment:
If `new_str` is `nullptr`, the return should be of size 0. I'm not sure what `std::string_view::empty` returns if it's `{nullptr, 12}`.